src/System/Process/Read.hs

```diff
@@ -349,17 +349,32 @@ getEnvOverride platform =
           mkEnvOverride platform
         . Map.fromList . map (T.pack *** T.pack)
 
+data PathException = PathsInvalidInPath [FilePath]
+    deriving Typeable
+
+instance Exception PathException
+instance Show PathException where
+    show (PathsInvalidInPath paths) = unlines $
+        [ "Would need to add some paths to the PATH environment variable \
+          \to continue, but they would be invalid because they contain a \
+          \colon."
+        , "Please fix the following paths and try again:"
+        ] ++ paths
+
 -- | Augment the PATH environment variable with the given extra paths.
-augmentPath :: [FilePath] -> Maybe Text -> Text
+augmentPath :: MonadThrow m => [FilePath] -> Maybe Text -> m Text
 augmentPath dirs mpath =
-    T.intercalate (T.singleton FP.searchPathSeparator)
-    $ map (T.pack . FP.dropTrailingPathSeparator) dirs
-   ++ maybe [] return mpath
+  do let illegal = filter (':' `elem`) dirs
```

Comment:
Uh oh; I suspect not! I'll try to make sure and/or figure out what to do instead.